Assets/Scripts/Screeps3D/RoomObjects/Views/ControllerView.cs

```diff
@@ -7,137 +7,122 @@ public class ControllerView : MonoBehaviour, IObjectViewComponent, IMapViewCompo
     {
         public const string Path = "Prefabs/RoomObjects/controller";
 
-        [SerializeField] private Renderer _rend = default;
-        [SerializeField] private Renderer _playerRend = default;
+        [SerializeField] private Renderer _badge = default;
+        [SerializeField] private Renderer _core = default;
+        [SerializeField] private Renderer _l1 = default;
+        [SerializeField] private Renderer _l2 = default;
+        [SerializeField] private Renderer _l3 = default;
+        [SerializeField] private Renderer _l4 = default;
+        [SerializeField] private Renderer _l5 = default;
+        [SerializeField] private Renderer _l6 = default;
+        [SerializeField] private Renderer _l7 = default;
+        [SerializeField] private Renderer _l8 = default;
+        [SerializeField] private ScaleVisibility _progress = default;
         [SerializeField] private ScaleVisibility _vis = default;
         [SerializeField] private Collider _collider = default;
         [SerializeField] private ParticleSystem _ps = default;
-
-        private Texture2D _texture;
-        private Color _controllerWhite;
         private Controller _controller;
-        private Material _psMaterial;
-        private bool _isMyReservation;
+        private Ownership _ownership;
+        enum Ownership {
+            Me,
+            Enemy,
+            None
+        }
+        
+        private void setReservation() {
+            if(_controller?.ReservedBy?.Badge != null) {
+                _badge.materials[0].SetColor("EmissionColor", new Color(0.7f, 0.7f, 0.7f, 1f));
+                _badge.materials[0].SetTexture("EmissionTexture", _controller.Owner.Badge);
+                _badge.materials[0].SetFloat("EmissionStrength", 5);
+                _ownership = _controller.ReservedBy.UserId.Equals(Screeps_API.ScreepsAPI.Me.UserId) ? Ownership.Me : Ownership.Enemy;
+            }
+        }
 
-        private Color _myColor;
-        private Color _enemyColor;
-        public void Init()
-        {
-            InitTexture();
+        private void setOwnership() {
+            if (_controller?.Owner?.Badge != null) {
+                _badge.materials[0].SetColor("EmissionColor", new Color(0.7f, 0.7f, 0.7f, 1f));
+                _badge.materials[0].SetTexture("EmissionTexture", _controller.Owner.Badge);
+                _badge.materials[0].SetFloat("EmissionStrength", 5f);
+                _ownership = _controller.Owner.UserId.Equals(Screeps_API.ScreepsAPI.Me.UserId) ? Ownership.Me : Ownership.Enemy;
+            }
+        }
```

Comment:
Color should probably only be initialized once, and not everytime we set ownership, should probably also detect owner, and only update if the owner changes